packages/qwik/src/optimizer/src/plugins/vite.ts

```diff
@@ -665,14 +665,7 @@ export function qwikVite(qwikViteOpts: QwikVitePluginOptions = {}): any {
           );
         };
         // TODO: Removed the "__qwikCityNew" condition in V3
-        const isNEW =
-          (globalThis as any).__qwikRouterNew === true ||
-          (globalThis as any).__qwikCityNew === true;
-        if (isNEW) {
-          return plugin;
-        } else {
-          return plugin();
-        }
+        return plugin;
```

Comment:
can you look in the commit history to see why this was added?

Are you sure that the function gets executed? Can you add a console.log() inside it to check?

Reply:
If it directly returns a function, that function will be invoked by Vite at the very end

Reply:
@JerryWu1234 if the comment says "remove in V3" it means that we deprecate it in V2 and then in V3 we'll remove this functionality as a breaking change. 

But for now, IIRC this was added to prevent backwards compatability issues, are you saying this has no effect?

can we just check for "library mode" instead and return the plugin in that mode specifically?